Framework/NexusGeometry/src/NexusGeometrySave.cpp

```diff
@@ -339,6 +357,7 @@ inline void writePixelData(H5::Group &grp,
       if (compInfo.hasValidShape(*it)) {
         auto pixelShapeInfo = compInfo.shape(*it).shapeInfo();
         auto type = pixelShapeInfo.shape();
+        // only write if shape is cylinder
```

Comment:
If we are encountering shapes that are no cylinders and are of type `CSGObject` I think that information should be logged. See `NexusGeometryParser` that uses a logger. If this seems too complex at this stage, please write an issue for this and assign it to yourself to handle later.